staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -319,6 +323,10 @@ func (u *Unstructured) GetAnnotations() map[string]string {
 }
 
 func (u *Unstructured) SetAnnotations(annotations map[string]string) {
+	if annotations == nil {
```

Comment:
or even len()==0 ?

Reply:
:+1: Wondering if we should also do the same for empty strings, etc?  Example: `SetName(name string)`.

`Name` has the omitEmpty tag and from https://golang.org/pkg/encoding/json/:

```
The "omitempty" option specifies that the field should be omitted from the encoding if the field has an empty value, defined as false, 0, a nil pointer, a nil interface value, and any empty array, slice, map, or string. 
```

Reply:
Btw if we do this, bonus point is that `kubectl get --export` will strip fields properly. Currently, it shows the fields for `Unstructured` objects (but with empty values).

Reply:
Updated. ptal, thanks.

This fixes `kubectl get --export` issue too now.